lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -46,6 +46,18 @@ def visit(node)
 
         name = fully_qualify_name(node.target.location.slice)
         add_constant(node, name)
+      when YARP::MultiWriteNode
+        value = node.value
+        values = value.is_a?(YARP::ArrayNode) ? value.elements : [value]
+
+        node.targets.each_with_index do |target, i|
```

Comment:
This is unfortunately only going to work in a subset of cases, and going to be wrong in many. Consider these:

```ruby
Foo, *, Bar = baz
Foo, = bar
(Foo, Bar) = baz

Foo, Bar = *baz, qux
Foo, Bar = [baz, *qux]
```

All of these are wrong at the moment, and these are just off the top of my head. I'm sure there are more edge cases here to consider.

If you want to handle the subset of cases that it looks like you're trying to, you will need to validate that:

* the RHS is an array literal with no splats
* the LHS has no parentheses or splats
* the number of targets and values lines up 1:1

Reply:
We have both of those cases in our tests. Unless I'm missing something, they should be working as we expect.

The first one is not a constant target, so all we do is check if the right hand side is a constant literal. The second case, `Foo` will be an alias to `Baz` and `Bar` will be `nil`, which works in the current setup.

Reply:
No I mean in this case `Foo = 1` and `Bar = 2`. `Foo` is not equal to `Baz` in this case.

Reply:
Oh, sorry. I had our test style in mind when reading the example and thought each line was a different case. I now see what you meant to say.

Indeed, there's no way to handle proper destructuring like that unless it's an array literal. I changed the code so that in your example, `Baz`, `Foo` and `Bar` and all considered constants - since the right hand side is not an array literal.

Reply:
I think the implementation is conservative enough. I think it's significantly more likely that we'd miss constants rather than adding confusing results. We only consider the right hand side if it's an array literal with no splats - and we only do so to check if there's a constant alias inside.

For all other cases, we ignore the right hand side and just remember where the constant was declared, which allows us to provide definition, hover, completion, etc.

I would rather start like this and deal with corner cases we encounter, rather than miss the constant declarations.

Reply:
Currently you're not considering the right hand side only if it's an array literal. Consider:

```ruby
foo = bar, baz
```

that gives you an array node on the RHS. If you want to only consider array literals you need to check if `opening_loc` is set on it.